nemo/collections/asr/models/msdd_models.py

```diff
@@ -1408,8 +1417,114 @@ def run_overlap_aware_eval(
                 out_rttm_dir=self.out_rttm_dir,
             )
             output = score_labels(
-                self.AUDIO_RTTM_MAP, all_reference, all_hypothesis, collar=collar, ignore_overlap=ignore_overlap,
+                rttm_map, all_reference, all_hypothesis, collar=collar, ignore_overlap=ignore_overlap,
             )
             outputs.append(output)
         logging.info(f"  \n")
         return outputs
+
+    @classmethod
+    def from_pretrained(
+        cls,
+        diar_model_name: str = 'diar_msdd_telephonic',
+        vad_model_name: str = 'vad_multilingual_marblenet',
+        speaker_model_name: str = 'titanet_large',
+        domain: Optional[str] = None,
```

Comment:
unsure exactly what is against the convention; specifying a default model?

Reply:
I meant domain.

Reply:
Signature of from pretrained is supposed to be consistent across nemov- extra args like domain etc usually are not supposed to be used. Also it's not device=ste but map_location=str for Pytorch API. 

If domain is needed, since this is custom code anyway and only mimics from_pretrained, I suppose there's no choice but to keep it. But ideally I would incorporate it directly into your from pretrained name itself - for example stt_en_conformer_ctc_large_ls is completely descriptive - speech to text model, English language, Conformer architecture, CTC loss, Large model size, Librispeech dataset

Reply:
Agree that `domain` is out the norm.

Also agreement that the parameters should actually be stored with the model, however because this is still evolving (and hopefully in the future unnecessary) we still need to set these parameters somehow. Any alternatives to passing a string for picking the domain specific parameters? @tango4j @nithinraok @titu1994

Reply:
My view is since we are passing model names then domain doesn;t matter as those models define the type of domain, (like msdd telephonic) also msdd model itself have speaker_model and configs of multi-scale (as msdd is trained on those scale types)  and those parameters are mostly model specific. So lets keep it based on general model [config](https://github.com/NVIDIA/NeMo/blob/main/examples/speaker_tasks/diarization/conf/inference/diar_infer_general.yaml). But then general config doesn;t work well with telephonic msdd? (@tango4j ) 

So my main question is: If msdd model is telephonic, domain doesn;t matter as its only tuned for telephonic audio, and setting domain externally might actually hurt model performance as msdd model for meeting isn;t released yet. 
Does this support offline clustering without msdd?

Reply:
@nithinraok I haven't trained general model config yet, because of the VAD issues for the overall diarization performance. Currently it can only use "diar_telephonic_msdd" only. Can't use general model config now. 

Like @nithinraok said, currently, domain argument cannot be doing anything. Until I upload models for other domains.

Reply:
Taejin, have you checked performance of msdd_telephonic on clean meeting type of audio? If its bad then we need to support offline clustering without overlap detection for good performance wrt domain. 
https://github.com/NVIDIA/NeMo/blob/4804f314116cac841539aedcceec2a55e0a5a412/nemo/collections/asr/models/clustering_diarizer.py#L390 should work. So we should think of a parent class that can subclass both offline and neural diarizers to be able to diarize incoming audio.

Reply:
@nithinraok let's focus on building this API first. I will work on releasing general model and meeting model. 

Also if we tune for meeting domain, then it is also bad for NIST-SRE-2K type of dataset too. There should be a general VAD, general scale settings and everything.

Reply:
yes, no hurry on releasing MSDD models, but I am thinking if we should support offline speaker diarization without overlap for non-telephonic incoming audio (general). That will make this API complete and usable without restriction on incoming audio. 

yes, as you said we can do clustering diarizer setup for time being. cc: @SeanNaren

Reply:
@nithinraok Letting people choose clustering/MSDD is too much of information and makes us lose the point of making one-line API. So let's use MSDD only when domain='telephonic' and put clustering diarizer instead for other domains. I will publish diar_msdd_meeting shortly, then switch it to support OvlDetection.

Reply:
While I agree, there's no need for that level of rigidity. ASR handles this with "change decoding strategy" that is a specialized class that is a middle layer betwee model output and metric calculation - taking up the tasks of decoding the models output from and into various forms. 

For example while models output is always logits or logprobs, we may want to do beam or greedy decoding so we make the decoding logical flexible and dynamic (changeable at runtime). You could take a similar approach in the future

Reply:
For a first iteration, I've removed domain. We can just specify the model(s) to load